app/controllers/media_controller.rb

```diff
@@ -52,6 +52,12 @@ def download
     add_link_header(asset)
     add_frame_header
     proxy_to_s3_via_nginx(asset)
+  rescue Mongoid::Errors::DocumentNotFound => e
+    if requested_from_draft_assets_host? && !authenticated?
+      authenticate_user!
+    else
+      raise e
```

Comment:
Small nitpick, but to re-raise an exception you can just have `raise` on its own.